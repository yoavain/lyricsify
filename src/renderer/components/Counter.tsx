import * as React from "react";
import { useState } from "react";

require("./Counter.scss");
const redCubeImg = require("./RedCube.jpg");

const Counter: React.FunctionComponent = () => {
    const [counter, setCounter] = useState(0);

    return (
        <div className="counter">
            <p>
                <img src={redCubeImg} />
            </p>
            <p id="counter-value">Current value: {counter}</p>
            <p>
                <button id="increment" onClick={() => setCounter(counter + 1)}>
                    Increment
                </button>
                <button id="decrement" onClick={() => setCounter(counter - 1)}>
                    Decrement
                </button>
            </p>
        </div>
    );
};

export default Counter;
