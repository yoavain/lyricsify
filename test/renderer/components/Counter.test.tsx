import * as React from "react";
import * as renderer from "react-test-renderer";

import Counter from "../../../src/renderer/components/Counter";

describe("Counter component", () => {
    it("renders correctly", () => {
        const tree = renderer.create(<Counter />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
