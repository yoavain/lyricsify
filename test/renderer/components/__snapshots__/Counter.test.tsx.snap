// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component renders correctly 1`] = `
<div
  className="counter"
>
  <p>
    <img
      src="test-file-stub"
    />
  </p>
  <p
    id="counter-value"
  >
    Current value: 
    0
  </p>
  <p>
    <button
      id="increment"
      onClick={[Function]}
    >
      Increment
    </button>
    <button
      id="decrement"
      onClick={[Function]}
    >
      Decrement
    </button>
  </p>
</div>
`;
